kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyDeploymentDependentResource.java

```diff
@@ -121,13 +118,7 @@ private static Map<String, String> deploymentSelector(KafkaProxy primary) {
     }
 
     public static Map<String, String> podLabels(KafkaProxy primary) {
-        Map<String, String> labelsFromSpec = Optional.ofNullable(primary.getSpec()).map(KafkaProxySpec::getPodTemplate)
-                .map(PodTemplateSpec::getMetadata)
-                .map(ObjectMeta::getLabels)
-                .orElse(Map.of());
-        Map<String, String> result = new LinkedHashMap<>(labelsFromSpec);
-        result.putAll(standardLabels(primary));
-        return result;
+        return standardLabels(primary);
```

Comment:
Does this mean we are just applying our labels and those will merge with user supplied labels. We have no mechanism to add default labels, right?

Reply:
Hmmm. We probably need a solution for that across all the generated resource in the 0.13+ time frame.

Reply:
yep plenty of options, we could add `infrastructure.all.labels` to supply custom labels to apply across the board, or we could have some option like `infrastructure.inheritLabels` to apply all the users labels from KafkaProxy.metadata.label to all resources

Feel like we need a proposal for this and the PodSpec ideas about how users can influence the proxy deployment/pod/container.